pyathena/sqlalchemy_athena.py

```diff
@@ -372,6 +439,16 @@ def get_schema_names(self, connection, **kw):
                 """
         return [row.schema_name for row in connection.execute(query).fetchall()]
 
+    @reflection.cache
+    def _get_table(self, connection, table_name, schema=None, **kw):
+        raw_connection = self._raw_connection(connection)
+        schema = schema if schema else raw_connection.schema_name
+        with raw_connection.connection.cursor() as cursor:
+            try:
+                return cursor._get_table_metadata(table_name, schema_name=schema)
+            except pyathena.error.OperationalError as exc:
+                raise NoSuchTableError(table_name) from exc
```

Comment:
It may not be the case that OperationalError is all about missing tables, you may need to check the API error response.